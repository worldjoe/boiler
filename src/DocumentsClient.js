class DocumentsClient {

  async getDocument(id) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, { method: 'GET'});
    const documentResponse = await response.json();
    return documentResponse;
  }

  async createDocument(content) {
    const response = await fetch("https://localhost:8443/v1/documents/", {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({'content' : content}),
    });
    const documentResponse = await response.json();
    return documentResponse;
  }

  async deleteDocument(id) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    //    const documentResponse = await response.json();
    return response;
  }

  async getAllDocuments() {
    console.log("joey in getAllDocuments");
    const response = await fetch("https://localhost:8443/v1/documents/", { method: 'GET'});
    console.log("joey in getAllDocuments response");
    const documentResponse = await response.json();
    console.log("joey in getAllDocuments response getting json");
    return documentResponse;
  }

  async updateDocument(id, content) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, {
      method: 'PUT', // or 'POST'
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({'content' : content}),
    });
    const documentResponse = await response.json();
    return documentResponse;
  }
}
export default DocumentsClient;
