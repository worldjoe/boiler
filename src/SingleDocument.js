import React, { Component } from "react";
import DocumentsClient from './DocumentsClient.js';
import { connect } from 'react-redux';
import { addDocumentAsync, deleteDocumentAsync, editDocumentAsync } from './redux/actions';

class SingleDocument extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    //    console.log("Joey props");
    //console.log(this.props.document.id);
    this.state = (props.document) ? props.document : { 'content': ''};
    //this.state = props;
  }

  handleChange = event => {
    this.setState({ content: event.target.value });
  };

  handleAddDocument = () => {
    this.props.addDocumentAsync(this.state.content);
    this.setState({ content : "" });
  };

  handleDeleteDocument = () => {
    this.props.deleteDocumentAsync(this.state.id);
    this.setState({ content : "" });
  };

  handleEditDocument = () => {
    this.props.editDocumentAsync(this.state.id, this.state.content);
    this.setState({ content : this.state.content });
  };

  render() {
  return (
    <div className="SingleDocument" id={this.state.id}>
      <form>
        {this.state.id ? (
          <React.Fragment>
        <label>id</label>
        <input readOnly value={this.state.id}/></React.Fragment>) : <div/>}

        <label>content</label>
        <input value={this.state.content} onChange={this.handleChange}/>
        <input type='button'  onClick={this.state.id ? this.handleEditDocument : this.handleAddDocument } value={this.state.id ? ('submit') : 'new'}/>
        {this.state.id ? (
          <input type='button'  onClick={this.handleDeleteDocument} value='delete'/>
        ) : <div/>}
      </form>
    </div>
  );
  }
}

export default connect(null, {addDocumentAsync, deleteDocumentAsync, editDocumentAsync} ) (SingleDocument);
