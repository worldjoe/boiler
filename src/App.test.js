/* Dependencies */
import React from 'react'
import renderer from 'react-test-renderer';
import App from './App.js'
import * as DocumentsClient from './DocumentsClient.js';
import SingleDocument from './SingleDocument.js';
jest.mock('./DocumentsClient');

describe('test App', () => {

  test ("donothing", () => {

  });
  /*
  test("no documents", () => {
    const noDocumentsApp = renderer.create(
      <App/>
    );
    let tree = noDocumentsApp.toJSON();
    expect(tree).toMatchSnapshot();
  })


  function sortStack(stack)  {
    if (stack.length === 0) {
      return stack;
    }
    let tempStack=[];
    let top = stack.pop();
    let newTop = stack.pop();
    while (newTop != null) {
      if (top > newTop) {
        tempStack.push(newTop);
      } else {
        if (tempStack.length>0)
          stack.push(...tempStack);
        sortStack(stack);
        stack.push(top);
        stack.push(newTop);
        return stack;
      }
      newTop = stack.pop();
    }
    stack.push(top);
    stack.push(...tempStack)
    if (stack.length === 1) {
      return stack;
    }
    return sortStack(stack);
  }

  test("sort test", () => {

    let stack = [34, 3];
    let sortedStack = sortStack(stack);
    expect(sortedStack).toEqual([3,34]);
  });

  test("mid sort test", () => {
    let newstack = [34, 3, 31];
    let sortedStack = sortStack(newstack);
    expect(sortedStack).toEqual([3, 31, 34]);
  });

  test("mid mid sort test", () => {
    let stack = [34, 3, 31, 98 ];
    let sortedStack = sortStack(stack);
    expect(sortedStack).toEqual([3, 31, 34, 98])
  });

  test("one document", () => {
    let myDocuments = [{'id': '123', 'content': 'daft punk'}];
    const noDocumentsApp = renderer.create(
      <App documents={myDocuments}/>
    );
    let tree = noDocumentsApp.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("one document and create one", () => {
    let myDocuments = [{'id': '123', 'content': 'daft punk'}];
    const noDocumentsApp = renderer.create(
      <App documents={myDocuments}/>
    );
    let tree = noDocumentsApp.toJSON();
    expect(tree).toMatchSnapshot();
    myDocuments = [{'id': '123', 'content': 'daft punk'}, {'id': '3245', 'content': 'stronger'}];
    noDocumentsApp.getInstance().setState({documents: myDocuments});
    tree = noDocumentsApp.toJSON();
    expect(tree).toMatchSnapshot();

  })
  */
})