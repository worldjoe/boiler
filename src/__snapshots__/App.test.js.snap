// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App no documents 1`] = `
<div
  className="App"
>
  <div>
    fallback
  </div>
  <div>
    <div
      className="SingleDocument"
    >
      <form>
        <div />
        <label>
          content
        </label>
        <input
          onChange={[Function]}
          value=""
        />
        <input
          onClick={[Function]}
          type="button"
          value="new"
        />
        <div />
      </form>
    </div>
  </div>
</div>
`;

exports[`test App one document 1`] = `
<div
  className="App"
>
  <div
    className="SingleDocument"
    id="123"
  >
    <form>
      <label>
        id
      </label>
      <input
        readOnly={true}
        value="123"
      />
      <label>
        content
      </label>
      <input
        onChange={[Function]}
        value="daft punk"
      />
      <input
        onClick={[Function]}
        type="button"
        value="submit"
      />
      <input
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </form>
  </div>
  <div>
    <div
      className="SingleDocument"
    >
      <form>
        <div />
        <label>
          content
        </label>
        <input
          onChange={[Function]}
          value=""
        />
        <input
          onClick={[Function]}
          type="button"
          value="new"
        />
        <div />
      </form>
    </div>
  </div>
</div>
`;

exports[`test App one document and create one 1`] = `
<div
  className="App"
>
  <div
    className="SingleDocument"
    id="123"
  >
    <form>
      <label>
        id
      </label>
      <input
        readOnly={true}
        value="123"
      />
      <label>
        content
      </label>
      <input
        onChange={[Function]}
        value="daft punk"
      />
      <input
        onClick={[Function]}
        type="button"
        value="submit"
      />
      <input
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </form>
  </div>
  <div>
    <div
      className="SingleDocument"
    >
      <form>
        <div />
        <label>
          content
        </label>
        <input
          onChange={[Function]}
          value=""
        />
        <input
          onClick={[Function]}
          type="button"
          value="new"
        />
        <div />
      </form>
    </div>
  </div>
</div>
`;

exports[`test App one document and create one 2`] = `
<div
  className="App"
>
  <div
    className="SingleDocument"
    id="123"
  >
    <form>
      <label>
        id
      </label>
      <input
        readOnly={true}
        value="123"
      />
      <label>
        content
      </label>
      <input
        onChange={[Function]}
        value="daft punk"
      />
      <input
        onClick={[Function]}
        type="button"
        value="submit"
      />
      <input
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </form>
  </div>
  <div
    className="SingleDocument"
    id="3245"
  >
    <form>
      <label>
        id
      </label>
      <input
        readOnly={true}
        value="3245"
      />
      <label>
        content
      </label>
      <input
        onChange={[Function]}
        value="stronger"
      />
      <input
        onClick={[Function]}
        type="button"
        value="submit"
      />
      <input
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </form>
  </div>
  <div>
    <div
      className="SingleDocument"
    >
      <form>
        <div />
        <label>
          content
        </label>
        <input
          onChange={[Function]}
          value=""
        />
        <input
          onClick={[Function]}
          type="button"
          value="new"
        />
        <div />
      </form>
    </div>
  </div>
</div>
`;
